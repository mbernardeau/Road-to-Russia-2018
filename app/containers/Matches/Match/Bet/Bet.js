import React from 'react';

import { imgUrl } from 'components/Flag';
import Reversible from 'components/Reversible';

import PropTypes from 'prop-types';
import _ from 'lodash';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';

import styles from './Bet.scss';

/**
 * Render menu items once (from 0 to 10 goals)
 */
const menuItems = _.map(_.range(11), (n) => <MenuItem value={n} key={n} primaryText={`${n}`} />);

const selectValueStyle = {
  textAlign: 'center',
  width: '100%',
  color: 'white',
  fontSize: 'bold',
};

/**
 * Pure mini-component to render inner value of the select field choices
 * @param {number} value Value to render
 *
 * @return {React.ReactElement}
 */
const renderValue = (value) => (<div style={selectValueStyle}>{ value }</div>);

const flagStyles = (country) => ({
  backgroundImage: `linear-gradient(
                      rgba(0, 0, 0, 0.5),
                      rgba(0, 0, 0, 0.5)
                    ),
                    url(${imgUrl(country)})`,
  backgroundSize: 'cover',
});

const Bet = ({ team, betValue, onBetValueUpdated, direction }) => (
  <div style={flagStyles(team.code)}>
    <Reversible direction={direction} className={styles.bet}>
      <h2 className={styles.teamName}>{team.name}</h2>
      <div className={styles.selectContainer}>
        <SelectField
          style={{ width: 60, color: 'white' }}
          value={betValue}
          selectionRenderer={renderValue}
          onChange={onBetValueUpdated}
          menuItemStyle={{ textAlign: 'center', width: 60 }}
        >
          {menuItems}
        </SelectField>
      </div>
    </Reversible>
  </div>
);


Bet.propTypes = {
  direction: PropTypes.oneOf(['rtl', 'ltr']),
  team: PropTypes.shape({
    name: PropTypes.string.isRequired,
    code: PropTypes.string.isRequired,
  }),
  onBetValueUpdated: PropTypes.func,
  betValue: PropTypes.number,
};

export default Bet;
