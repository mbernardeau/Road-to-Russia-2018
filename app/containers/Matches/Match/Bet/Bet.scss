.countryFlag {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
}

.bet {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.teamName {
  text-align: center;
}

.selectContainer {
  display: flex;
  align-items: center;
}
