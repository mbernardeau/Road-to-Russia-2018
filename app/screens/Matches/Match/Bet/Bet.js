import React from 'react'

import Flag from 'components/Flag'

import PropTypes from 'prop-types'

import { map, range } from 'lodash'

import Select from 'material-ui/Select'
import { MenuItem } from 'material-ui/Menu'

const Bet = ({ team, betValue, onBetValueUpdated }) => (
  <div style={styles.bet}>
    <div style={styles.betTitle}>
      <Flag country={team.code} style={styles.flag} />
      <div style={styles.teamName}>{team.name}</div>
    </div>
    <div style={styles.selectContainer}>
      <Select
        type="number"
        value={betValue || ''}
        renderValue={renderValue}
        onChange={onBetValueUpdated}
      >
        {menuItems}
      </Select>
    </div>
  </div>
)

const styles = {
  bet: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'center',
  },

  betTitle: {
    display: 'flex',
  },

  teamName: {
    textAlign: 'center',
    fontSize: '1.3em',
    paddingLeft: '10px',
  },

  selectContainer: {
    display: 'flex',
    alignItems: 'center',
  },

  selectValueStyle: {
    textAlign: 'center',
    width: '100%',
  },

  flag: {
    height: '1.3em',
  },
}

/**
 * Render menu items once (from 0 to 10 goals)
 */
const menuItems = map(range(11), n => (
  <MenuItem value={n} key={n}>
    {n}
  </MenuItem>
))

/**
 * Pure mini-component to render inner value of the select field choices
 * @param {number} value Value to render
 *
 * @return {React.ReactElement}
 */
const renderValue = value => <div style={styles.selectValueStyle}>{value}</div>

Bet.defaultProps = {
  team: {},
}

Bet.propTypes = {
  team: PropTypes.shape({
    name: PropTypes.string.isRequired,
    code: PropTypes.string.isRequired,
  }),
  onBetValueUpdated: PropTypes.func.isRequired,
  betValue: PropTypes.number,
}

export default Bet
