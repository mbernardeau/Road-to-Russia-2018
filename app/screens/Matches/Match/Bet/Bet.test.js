import React from 'react'
import renderer from 'react-test-renderer'

import Bet from './Bet'

jest.mock('@material-ui/core/Select', () => 'Select')
jest.mock('@material-ui/core/MenuItem', () => 'MenuItem')
jest.mock('components/Flag', () => 'Flag')

describe('screens/Match/Bet', () => {
  const props = {
    team: {
      code: 'ru',
      name: 'Russie',
    },
    betValue: 1,
    onBetValueUpdated: jest.fn(),
  }

  it('renders correctly', () => {
    const tree = renderer.create(<Bet {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
