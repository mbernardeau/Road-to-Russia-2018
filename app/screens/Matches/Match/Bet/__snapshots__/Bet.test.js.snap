// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/Match/Bet renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <img
      alt="ru"
      className={undefined}
      src="IMAGE_MOCK"
      style={
        Object {
          "height": "1.3em",
        }
      }
    />
    <div
      style={
        Object {
          "fontSize": "1.3em",
          "paddingLeft": "10px",
          "textAlign": "center",
        }
      }
    >
      Russie
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
      }
    }
  >
    <Select
      onChange={[MockFunction]}
      renderValue={[Function]}
      type="number"
      value={1}
    >
      <MenuItem
        value={0}
      >
        0
      </MenuItem>
      <MenuItem
        value={1}
      >
        1
      </MenuItem>
      <MenuItem
        value={2}
      >
        2
      </MenuItem>
      <MenuItem
        value={3}
      >
        3
      </MenuItem>
      <MenuItem
        value={4}
      >
        4
      </MenuItem>
      <MenuItem
        value={5}
      >
        5
      </MenuItem>
      <MenuItem
        value={6}
      >
        6
      </MenuItem>
      <MenuItem
        value={7}
      >
        7
      </MenuItem>
      <MenuItem
        value={8}
      >
        8
      </MenuItem>
      <MenuItem
        value={9}
      >
        9
      </MenuItem>
      <MenuItem
        value={10}
      >
        10
      </MenuItem>
    </Select>
  </div>
</div>
`;
